pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,863 @@
     DASK_IMPORTED = False
 
 
+class StepsNowcaster:
+    def __init__(self, precip, velocity, timesteps, **kwargs):
+        # Store inputs and optional parameters
+        self.precip = precip
+        self.velocity = velocity
+        self.timesteps = timesteps
+        self.n_ens_members = kwargs.get("n_ens_members", 24)
+        self.n_cascade_levels = kwargs.get("n_cascade_levels", 6)
+        self.precip_thr = kwargs.get("precip_thr", None)
+        self.kmperpixel = kwargs.get("kmperpixel", None)
+        self.timestep = kwargs.get("timestep", None)
+        self.extrap_method = kwargs.get("extrap_method", "semilagrangian")
+        self.decomp_method = kwargs.get("decomp_method", "fft")
+        self.bandpass_filter_method = kwargs.get("bandpass_filter_method", "gaussian")
+        self.noise_method = kwargs.get("noise_method", "nonparametric")
+        self.noise_stddev_adj = kwargs.get("noise_stddev_adj", None)
+        self.ar_order = kwargs.get("ar_order", 2)
+        self.vel_pert_method = kwargs.get("vel_pert_method", "bps")
+        self.conditional = kwargs.get("conditional", False)
+        self.probmatching_method = kwargs.get("probmatching_method", "cdf")
+        self.mask_method = kwargs.get("mask_method", "incremental")
+        self.seed = kwargs.get("seed", None)
+        self.num_workers = kwargs.get("num_workers", 1)
+        self.fft_method = kwargs.get("fft_method", "numpy")
+        self.domain = kwargs.get("domain", "spatial")
+        self.extrap_kwargs = kwargs.get("extrap_kwargs", None)
+        self.filter_kwargs = kwargs.get("filter_kwargs", None)
+        self.noise_kwargs = kwargs.get("noise_kwargs", None)
+        self.vel_pert_kwargs = kwargs.get("vel_pert_kwargs", None)
+        self.mask_kwargs = kwargs.get("mask_kwargs", None)
+        self.measure_time = kwargs.get("measure_time", False)
+        self.callback = kwargs.get("callback", None)
+        self.return_output = kwargs.get("return_output", True)
+
+        # Additional variables for internal state management
+        self.fft = None
+        self.bp_filter = None
+        self.extrapolator_method = None
+        self.domain_mask = None
+        self.precip_cascades = None
+        self.gamma = None
+        self.phi = None
+        self.pert_gen = None
```

Comment:
```suggestion
        self.perturbation_generator = None
```
I'm not sure if this is the right name, what do you think?
If this works for you, too, then we should change it in the rest of the code as well.